test/e2e/kuma_deploy_kubernetes_test.go

```diff
@@ -0,0 +1,124 @@
+package e2e
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Kong/kuma/test/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = BeforeEach(func() {
+
+})
+
+var _ = Describe("Test K8s deployment with `kumactl install control-plane`", func() {
+
+	It("Deploy on Single K8s cluster and verify the Kuma CP REST API is accessible", func(done Done) {
+		clusters, err := framework.NewK8sClusters(
+			[]string{framework.Kuma1},
+			framework.Silent)
+		Expect(err).ToNot(HaveOccurred())
+		c := clusters.GetCluster(framework.Kuma1)
+
+		err = c.DeployKuma()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = c.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+
+		logs, err := c.GetKumaCPLogs()
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Println(logs)
```

Comment:
2 ideas, but not blocking for this PR:
* pass `os.Stdout` (or so) so we have logs for the full test including deleting, because after this line we won't see logs
* Print logs only if test fails so it's easier to debug test but stdout is not flooded with logs

Reply:
Yep, this was kind of a demo that we can get the logs in case we need them. I am not sure what would be best though.